src/app/modules/item/services/item-task.service.ts

```diff
@@ -0,0 +1,120 @@
+import { LocaleService } from 'src/app/core/services/localeService';
+import { openNewTab, replaceWindowUrl } from 'src/app/shared/helpers/url';
+import { FullItemRoute, itemRoute } from 'src/app/shared/routing/item-route';
+import { ItemRouter } from 'src/app/shared/routing/item-router';
+import { Task, TaskPlatform } from '../task-communication/task-proxy';
+import { TaskParamsValue } from '../task-communication/types';
+import { ItemTaskAnswerService } from './item-task-answer.service';
+import { ItemTaskInitService } from './item-task-init.service';
+import { ItemTaskViewsService } from './item-task-views.service';
+
+@Injectable()
+export class ItemTaskService {
+  private error$ = merge(
+    this.answerService.error$,
+    this.viewsService.error$,
+  ).pipe(switchMap(error => throwError(() => error)));
+
+  readonly task$ = merge(this.initService.task$, this.error$);
+  readonly iframeSrc$ = this.initService.iframeSrc$;
+  get initialized(): boolean {
+    return this.initService.initialized;
+  }
+
+  readonly views$ = this.viewsService.views$;
+  readonly display$ = this.viewsService.display$;
+  readonly activeView$ = this.viewsService.activeView$;
+
+  private config$ = this.initService.config$;
+
+  constructor(
+    private initService: ItemTaskInitService,
+    private answerService: ItemTaskAnswerService,
+    private viewsService: ItemTaskViewsService,
+    private itemRouter: ItemRouter,
+    private itemNavigationService: ItemNavigationService,
+    private router: Router,
+    private localeService: LocaleService,
+  ) {}
+
+  configure(route: FullItemRoute, url: string, attemptId: string): void {
+    this.initService.configure(route, url, attemptId);
+  }
+
+  initTask(iframe: HTMLIFrameElement): void {
+    this.initService.initTask(iframe, task => this.bindPlatform(task));
+  }
+
+  showView(view: string): void {
+    this.viewsService.showView(view);
+  }
+
+  private bindPlatform(task: Task): void {
+    const platform = new TaskPlatform({
+      validate: (mode): Observable<void> => this.validate(mode).pipe(mapTo(undefined)),
+      getTaskParams: (): Observable<TaskParamsValue> =>
+        of({ minScore: -3, maxScore: 10, randomSeed: 0, noScore: 0, readOnly: false, options: {} }),
+      updateDisplay: (display): Observable<void> => {
+        this.viewsService.updateDisplay(display);
+        return EMPTY;
+      },
+      showView: (view): Observable<void> => {
+        this.viewsService.showView(view);
+        return EMPTY;
+      },
+      openUrl: (params): Observable<void> => {
+        if (typeof params === 'string') return this.navigateToItem(params);
+        if ('path' in params) return this.navigateToItem(params.path, params.newTab);
+        this.navigate(params.url, params.newTab);
+        return EMPTY;
+      }
+    });
+    task.bindPlatform(platform);
+  }
+
+  private validate(mode: string): Observable<unknown> {
```

Comment:
Is it relevant to return an observable of `unknown`? Does any part of the code parse that?
And who does the subscribe to that? It looks like a function where the caller does not really care about the outcome, in which case subscribing directly here may make sense.
(I haven't re-read the context yet)

Reply:
The subscriber is the rx-adapted jschannel

There might be some improvements to do regarding the platform functions because some of them require a return result (like `getTaskParams()`) and others should return `void`

But the rx-jschannel adapter requires an observable to complete the jschannel transaction.

And technically, return `EMPTY` as we currently doing is not that safe either, because jschannel closes the transaction on the first **emitted** value

After saying all that, I think we should have a discussion around this and make changes in another dedicated PR

Reply:
As discussed, I created an issue to treat that subject in another PR: #701